### PR TITLE
Exposed new WPCOM service method to retrieve site verticals

### PR DESCRIPTION
**Pull request by stevebaranski**
### Description

Implements the service to fulfill the service implementation of [this issue](https://github.com/wordpress-mobile/WordPress-iOS/issues/10319).

### Testing

- Checkout the branch. Confirm that it builds & tests pass.
- For additional assurance, consider checking out the WPiOS branch `issue/10319-wordpresskit-wip` and verify that it builds and that tests pass.

### CODE REVIEW

File: WordPressKit/WordPressComServiceRemote+SiteVerticals.swift [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    /// Retrieves Verticals matching the specified criteria.
+    ///
+    /// - Parameters:
+    ///   - request:    the value object with which to compose the request.
+    ///   - completion: a closure including the result of the request for site verticals.
+    ///
+    func retrievalVerticals(request: SiteVerticalsRequest, completion: @escaping SiteVerticalsServiceCompletion) {
+
+        let endpoint = "verticals"
+        let path = self.path(forEndpoint: endpoint, withVersion: ._2_0)
```
Comment (yaelirub) [inline WordPressKit/WordPressComServiceRemote+SiteVerticals.swift]: Oh this is great! At my last workplace we had a lot of issues due to the lack of API versioning and now I see the APi versions for ServiceRemoteWordPressComRESTApiVersion. This is great.

File: WordPressKit/WordPressComServiceRemote+SiteVerticals.swift [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+                guard let self = self else {
+                    return
+                }
+
+                do {
+                    let response = try self.decodeResponse(responseObject: responseObject)
+                    completion(.success(response))
+                } catch {
+                    DDLogError("Failed to decode \([SiteVertical].self) : \(error.localizedDescription)")
```
Comment (yaelirub) [inline WordPressKit/WordPressComServiceRemote+SiteVerticals.swift]: Any reason why you're logging the localizedDescription here but not in encodeRequestParameters?

File: WordPressKitTests/SiteVerticalsRequestEncodingTests.swift [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        // When
+        let encoder = JSONEncoder()
+
+        XCTAssertNoThrow(try encoder.encode(request))
+        let encodedJSON = try! encoder.encode(request)
+
+        XCTAssertNoThrow(try JSONSerialization.jsonObject(with: encodedJSON, options: []))
+        let serializedJSON = try! JSONSerialization.jsonObject(with: encodedJSON, options: [])
+
+        if let _ = serializedJSON as? [String : AnyObject] {} else {
```
Comment (yaelirub) [inline WordPressKitTests/SiteVerticalsRequestEncodingTests.swift]: Can we do 
```suggestion
        guard let jsonDictionary = serializedJSON as [String : AnyObject] else {
            XCTFail("Failed to encode a proper JSON dictionary!")
        }
```
and remove 
```suggestion
       let jsonDictionary = serializedJSON as! [String : AnyObject]
```
?

File: WordPressKitTests/SiteVerticalsResponseDecodingTests.swift [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        let secondVertical = SiteVertical(identifier: "", title: "", isNew: MockValues.isNew)
+
+        XCTAssertNotNil(singleVerticalSUT)
+        XCTAssertNotEqual(singleVerticalSUT, secondVertical)
+    }
+
+    // MARK: Multiple Verticals
+
+    func testSiteParsingOfMultipleVerticalsWorksAsExpected() {
+        XCTAssertTrue(!verticalsSUT.isEmpty)
```
Comment (yaelirub) [inline WordPressKitTests/SiteVerticalsResponseDecodingTests.swift]: why do we need to test that its not empty and that it is equal 5? shouldn't equal 5 be enough?

**Comment by yaelirub** [comment]
@stevebaranski , thank you so much for including me in this! Such a well written PR!

File: WordPressKitTests/SiteVerticalsResponseDecodingTests.swift (hunk 4 above)
Comment (yaelirub) [reply WordPressKitTests/SiteVerticalsResponseDecodingTests.swift]: 👍

File: WordPressKitTests/SiteVerticalsRequestEncodingTests.swift (hunk 3 above)
Comment (yaelirub) [reply WordPressKitTests/SiteVerticalsRequestEncodingTests.swift]: 👍 Do you think we should do the same in line 53?
